clarify ssh tunnel dbt connection error (#4913)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/connect-data-platform/connect-redshift-postgresql-alloydb.md

The Bastion server should now be ready for dbt Cloud to use as a tunnel into the Redshift environment.

## Configuration

To optimize performance with data platform-specific configurations in dbt Cloud, refer to [Redshift-specific configuration](/reference/resource-configs/redshift-configs).

To grant users or roles database permissions (access rights and privileges), refer to the [Redshift permissions](/reference/database-permissions/redshift-permissions) page or [Postgres permissions](/reference/database-permissions/postgres-permissions) page.

## FAQs

<detailsToggle alt_header="Database Error - could not connect to server: Connection timed out">

> When setting up a database connection using an SSH tunnel, you typically need these parts:

Reviewer: Unless we're going to also present a scenario where they wouldn't need all three, we should be firm in the requirements:
```suggestion
When setting up a database connection using an SSH tunnel, you need the following components:
```

> - A Load Balancer (like ELB or NLB) to manage traffic.

Reviewer: I think this may be better served keeping the initialisms spelled out first.

> - A Load Balancer (like ELB or NLB) to manage traffic.

Reviewer: ```suggestion
- A load balancer (like ELB or NLB) to manage traffic.
```

- A bastion host (or jump server) that runs the SSH protocol, acting as a secure entry point.
- The database itself (such as a Redshift cluster).

> dbt Cloud uses an SSH tunnel to connect through the Load Balancer to the database. This tunnel is set up at the start of any dbt job run. If the tunnel drops, the job fails.

Reviewer: ```suggestion
dbt Cloud uses an SSH tunnel to connect through the load balancer to the database. This connection is established at the start of any dbt job run. If the tunnel connection drops, the job fails.
```

Tunnel failures usually happen because:

> - The SSH service stops if there's no activity for too long.

Reviewer: The SSH service is a much broader term than SSH daemon. It might be better to be specific here unless someone on our team suggested the change. 
```suggestion
- The SSH daemon times out if it's idle for too long.
```

> - The Load Balancer cuts off the connection if it's idle.

Reviewer: ```suggestion
- The load balancer cuts off the connection if it's idle.
```

- dbt Cloud tries to keep the connection alive by checking in every 30 seconds, and the system will end the connection if there's no response from the SSH service after 300 seconds. This helps avoid drops due to inactivity unless the Load Balancer's timeout is less than 30 seconds.

Bastion hosts might have additional SSH settings to disconnect inactive clients after several checks without a response. By default, it checks three times.

> To prevent premature disconnections, you can adjust the settings on the bastion host:

Reviewer: If these are recommended settings, we should state that, otherwise leave it a little more open ended for customers to configure what's right for them

> - `ClientAliveCountMax 10` &mdash; Allows more checks before deciding the client is inactive.

Reviewer: ```suggestion
- `ClientAliveCountMax ` &mdash; Configures the number of checks before deciding the client is inactive. For example, `ClientAliveCountMax 10` checks 10 times.
```

> - `ClientAliveInterval 30` &mdash; Checks for client activity every 30 seconds.

Reviewer: ```suggestion
- `ClientAliveInterval` &mdash; Configures when to check for client activity. For example, `ClientAliveInterval 30` checks every 30 seconds.
```

> These adjustments ensure that inactive SSH clients are disconnected after about 300 seconds, reducing the chance of tunnel failures.

Reviewer: ```suggestion
The example adjustments ensure that inactive SSH clients are disconnected after about 300 seconds, reducing the chance of tunnel failures.
```
